File: core/src/main/java/com/mercer/annotate/http/Serialization.java
package com.mercer.annotate.http;

import com.mercer.core.Converter;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * @author :Mercer
 * @Created on 2025/01/03.
 * @Description: 序列化
 */
@Target({ElementType.TYPE, ElementType.METHOD})
@Retention(RetentionPolicy.RUNTIME)
public @interface Serialization {

    @SuppressWarnings("rawtypes")
    Class<? extends Converter.Factory> value();

}

File: core/src/main/java/com/mercer/annotate/http/Serialization.kt
package com.mercer.annotate.http

import com.mercer.core.Converter
import kotlin.reflect.KClass

/**
 * @author :Mercer
 * @Created on 2024/11/27.
 * @Description:
 *   序列化
 */
annotation class Serialization(val value: KClass<out Converter.Factory<*>>)